rokka/sourceimages.go

```diff
@@ -122,3 +126,36 @@ func (c *Client) CreateSourceImageWithMetadata(org, name string, data io.Reader,
 
 	return result, err
 }
+
+func (c *Client) DynamicMetadataResponseHandler(v interface{}, resp *http.Response) error {
+	if resp.StatusCode == 204 {
+		dynamicMetadataResponse, ok := v.(*PutDynamicMetadataResponse)
+
+		if ok {
+			dynamicMetadataResponse.Location = resp.Header.Get("Location")
+
+			return nil
+		}
+	}
+
+	return StatusCodeError{
+		resp.StatusCode,
+		[]byte{},
```

Comment:
the body still needs to parsed and returned in order for the callee to be able to inspect/log what the error response body contained. Hence my request that status code handling should be done for all, not only if response handler is not set.